kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/EnvelopeEncryptionFilterIT.java

```diff
@@ -248,6 +257,77 @@ void produceAndConsumeEncryptedAndPlainTopicsAtSameTime(KafkaCluster cluster, To
         }
     }
 
+    /**
+     * Test that ensures that the record offsets returned by the broker are faithfully relayed to the client.
+     * @param cluster underlying kafka cluster
+     * @param compactedTopic topic configured for compaction.
+     * @param directConsumer consumer connected directly to the underlying kafka cluster
+     * @param testKmsFacade kms facade
+     * @throws Exception exception
+     */
+    @TestTemplate
+    @SuppressWarnings("java:S2925")
+    void offsetFidelity(@BrokerConfig(name = "log.cleaner.backoff.ms", value = "500") @BrokerConfig(name = "log.retention.check.interval.ms", value = "500") KafkaCluster cluster,
+                        @TopicConfig(name = "segment.ms", value = "250") @TopicConfig(name = "cleanup.policy", value = "compact") Topic compactedTopic,
```

Comment:
It's sad that the formatting makes this so unreadable, but not your fault.

Reply:
Is there a case for `@BrokerConfigs(keys = [], values = [])` and use matching array indexes to make it more readable? A bit ugly in theory but with a some bounds checking its probably good enough in practice?

Reply:
Something to make broker/topic/producer/consumer more succinct would be desirable.